doc/dev/release-procedure.md

```diff
@@ -0,0 +1,289 @@
+
+```shell
+screen /dev/ttyUSB2 921600
+```
+
+You should see output from the proximity demo:
+
+```
+proximity sensor example: Proximity is 0x12
+proximity sensor example: Proximity is 0x11
+proximity sensor example: Proximity is 0x16
+proximity sensor example: Proximity is 0x19
+proximity sensor example: Proximity is 0x1a
+proximity sensor example: Proximity is 0x1c
+proximity sensor example: Proximity is 0x1c
+proximity sensor example: Proximity is 0x1a
+proximity sensor example: Proximity is 0x18
+```
+
+These values will change as you move your hand over the proximity sensor.
+
+### Snake demo
+
+Load snake demo and play snake:
+
+```shell
+# Copy to parent release dir
+cp ./build/cheriot/cheriot/release/snake_demo.slot1.uf2  ../snake_demo_vX.Y.slot1.uf2
+# Program onto the FPGA
+cp ../snake_demo_vX.Y.slot1.uf2 /path/to/SONATA/
+```
+
+Check that you see capability the exception LEDs light up and fade out when you hit the game boundaries ('tag' exception for top and left boundaries, 'bounds' exception for bottom and right boundaries).
+
+### RTOS test suite
+
+Run and build the CHERIoT RTOS test suite:
+
+```shell
+rm -r build .xmake
+xmake config -P ./cheriot-rtos/tests/ --board=sonata-prerelease
+xmake -P ./cheriot-rtos/tests/
+llvm-strip ./build/cheriot/cheriot/release/test-suite -o test-suite.strip
+uf2conv -b 0x00000000 -f 0x6ce29e60 test-suite.strip -co test-suite.slot1.uf2
+cp test-suite.slot1.uf2 /path/to/SONATA
+``` 
+
+Results are output on the terminal (see instructions above), you will want to open that before you run the test suite otherwise you may miss it!
+
+A successful run will output a large amount of text, it ends with something like:
+
+```
+Allocator test: fuzz i=0x0
+Allocator test: fuzz i=0x8
+Allocator test: fuzz i=0x10
+Allocator test: fuzz i=0x18
+Allocator test: fuzz i=0x20
+Allocator test: fuzz i=0x28
+Allocator test: fuzz i=0x30
+Allocator test: fuzz i=0x38
+Test runner: Allocator finished in 30477954 cycles
+Test runner: All tests finished in 42895559 cycles
+```
+
+## Make Release
+
+In your release directory from the procedures above you should have:
+ - `sonata-vX.Y.bit.slot1.uf2`
+ - `sonata_simple_demo_vX.Y.slot1.uf2`
+ - `snake_demo_vX.Y.slot1.uf2`
+
+Use these to create the GitHub release, remember to include the RP2040 UF2 and include appropriate release notes.
+
+### Versioning
+
+Choose a release number in the form "vX.Y" which is higher than the previous release.
+Create a tag in sonata-system:
+```shell
+# Create git tag
+git tag vX.Y
+# Push this tag to upstream
+git push --set-upstream upstream vX.Y
+```
+
+For the sonata-software repository we should create a branch:
+```shell
+git checkout -b vX.Y
+git push --setupstream upstream vX.Y
+```
+
+### Release notes
+
+Look for the previous tagged release and go through the commit history since then.
+Note down any major updates like the additions of an IP block and make a bulleted list.
+An example release notes looks something like this:
```

Comment:
In the release notes we should add a table that gives some top-level PPA figures from the FPGA build e.g. recreate the FPGA utilization table you see in Vivado and list the WNS per-clock (or at least WNS of the system clock, note this may be different from the top-level WNS!). We could also put the top-level power figure in there.

Reply:
Good idea! I've added an example list of interesting figures we might want.